### PR TITLE
various performance improvements

### PR DESCRIPTION
**Pull request by agourlay**
While profiling my application, I found a couple of hotspots that are straightforward to mitigate. 

This PR contains two types of fixes:
- fixes for things I have seen during profiling which I will annotate with a comment
- proactive fixes which should be beneficial 

Thank you for this nice library.

### CODE REVIEW

File: src/main/scala/sangria/slowlog/MetricRenderer.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ object MetricRenderer {
 class DefaultMetricRenderer(val unit: TimeUnit) extends MetricRenderer {
   def renderVariables[In: InputUnmarshaller](variables: In, names: Vector[String]) = {
     val iu = implicitly[InputUnmarshaller[In]]
-    val renderedVars = names.flatMap(name =>
+    val renderedVars = names.iterator.flatMap(name =>
```
Comment (agourlay) [inline src/main/scala/sangria/slowlog/MetricRenderer.scala:28]: found during profiling, no need to allocate a collection for reducing it with `mkString`

File: src/main/scala/sangria/slowlog/QueryMetrics.scala [hunk 2]
```diff
@@ -168,23 +168,24 @@ case class QueryMetrics(
       typeInfo: TypeInfo,
       typeMetrics: TrieMap[String, FieldMetrics],
       prefix: String = "")(implicit renderer: MetricRenderer): Vector[ast.Comment] = {
+    val previousParentType = typeInfo.previousParentType
```
Comment (agourlay) [inline src/main/scala/sangria/slowlog/QueryMetrics.scala:171]: found during profiling, `typeInfo.previousParentType` is actually an expensive `def`

File: src/main/scala/sangria/slowlog/QueryMetrics.scala [hunk 3]
```diff
@@ -279,7 +280,7 @@ case class QueryMetrics(
     else added
 
   def toComments(s: String): Vector[ast.Comment] =
-    s.split("\n").map(c => ast.Comment(c.trim)).toVector
+    s.split("\n").iterator.map(c => ast.Comment(c.trim)).toVector
```
Comment (agourlay) [inline src/main/scala/sangria/slowlog/QueryMetrics.scala:283]: found during profiling

File: src/main/scala/sangria/slowlog/SlowLog.scala [hunk 4]
```diff
@@ -96,7 +96,11 @@ class SlowLog(
     queryVal.update(path, ctx.parentType.name, ctx.field.name, success, fieldVal, System.nanoTime())
   }
 
-  def even[T](v: Vector[T]): Vector[T] = v.zipWithIndex.filter(_._2 % 2 == 0).map(_._1)
+  def even[T](v: Vector[T]): Vector[T] = v.iterator
```
Comment (agourlay) [inline src/main/scala/sangria/slowlog/SlowLog.scala:99]: found during profiling

**Review by yanns — APPROVED** [review]
💯

**Comment by yanns** [comment]
@agourlay could you please run the formatter?
```
[warn] /home/runner/work/sangria-slowlog/sangria-slowlog/src/main/scala/sangria/slowlog/QueryMetrics.scala isn't formatted properly!
[warn] /home/runner/work/sangria-slowlog/sangria-slowlog/src/main/scala/sangria/slowlog/SlowLog.scala isn't formatted properly!
[error] 2 files must be formatted
[error] (Compile / scalafmtCheck) 2 files must be formatted
```

**Comment by agourlay** [comment]
@yanns Thanks for the approval, formatting done!

**Pull request merged**